src/status_im2/contexts/chat/bottom_sheet_composer/mentions/view.cljs

```diff
@@ -0,0 +1,54 @@
+(ns status-im2.contexts.chat.bottom-sheet-composer.mentions.view
+  (:require
+    [react-native.platform :as platform]
+    [reagent.core :as reagent]
+    [status-im2.contexts.chat.bottom-sheet-composer.utils :as utils]
+    [utils.re-frame :as rf]
+    [react-native.core :as rn]
+    [react-native.reanimated :as reanimated]
+    [status-im2.common.contact-list-item.view :as contact-list-item]
+    [status-im2.contexts.chat.bottom-sheet-composer.mentions.style :as style]))
+
+(defn mention-item
+  [user _ _ {:keys [cursor-position input-ref]}]
+  [contact-list-item/contact-list-item
+   {:on-press (fn []
+                (let [new-cursor-pos (+ (count (:primary-name user)) @cursor-position)]
+                  (rf/dispatch [:chat.ui/select-mention nil user])
+                  (when platform/android?
+                    (reset! cursor-position new-cursor-pos)
+                    (reagent/next-tick #(when @input-ref
+                                          (.setNativeProps ^js @input-ref
+                                                           (clj->js {:selection {:start new-cursor-pos
+                                                                                 :end
+                                                                                 new-cursor-pos}})))))))}
+   user])
+
+(defn- f-view
+  [suggestions suggestions-atom props state animations max-height cursor-pos]
+  (let [opacity (reanimated/use-shared-value (if (seq suggestions) 1 0))
+        mentions-pos
+        (utils/calc-suggestions-position cursor-pos state animations max-height (count suggestions))]
+    (rn/use-effect
+     (fn []
+       (if (seq suggestions)
+         (reagent/next-tick #(reset! suggestions-atom suggestions))
+         (js/setTimeout #(reset! suggestions-atom suggestions) 300))
```

Comment:
I don't understand this part of the code. Why store the result of a subscription in an atom? There should be a better/simpler way.

Reply:
That's for the animation. The data is being rendered from the `suggestions-atom`. I want to reset the `suggestions-atom` after the animation is complete.

Reply:
Thanks for the explanation @OmarBasem. But if `suggestions` comes from `[:chat/mention-suggestions]`, why can't we dispatch an event to clear the data in the app db instead? I've never seen a case where we need to duplicate the results of a subscription in an atom, that's why this code raised my eyebrows. This doesn't seem correct.

Reply:
@ilmotta We are doing so. But the animations takes `300ms` to complete, so I need to store the data in a secondary storage.

Reply:
In `on-change-text` in `handlers` we are calling `(rf/dispatch [:mention/on-change-text text])` which handles whether there is suggestions or not. When the `@` is cleared, or a mention is selected` we clear app db. But I still need the data for animation. Hence storing it in an atom.

Reply:
> This doesn't seem correct (it does work, but doesn't seem like something we should ever do).

@ilmotta btw this is a common pattern in react-native. A lot of times the data needed to be in state to react on is not identical to the data we get from the app's storage. So we need to store a copy in state/atom. And React has `getDerivedStateFromProps` which does exactly that https://react.dev/reference/react/Component#static-getderivedstatefromprops. We can then have `lastest-data` from `app-db` or `props` in JS react, and `prev-data` in the component's local `state`

Reply:
Yes, this is very common. It's just that it took me a while to understand from the code, because it's mixed with the rest of the logic, and the name `suggestions-atom` doesn't say much.

There's usually a need for a hook often named `usePrevious`, kind of similar to what you're describing, but not exactly the same.

It'd be good to have abstractions for these things so we can keep the view sane. :+1: for the suggestion